src/section/mod.rs

```diff
@@ -208,6 +208,12 @@ impl Section {
         } else if expected_names.len() < crate::majority(current_names.len()) {
             warn!("ignore attempt to reduce the number of elders too much");
             vec![]
+        } else if expected_names.len() <= 1 {
```

Comment:
don't get this branch, this branch will only be reached when `current_names.len()` is 1 ?
which actually is handling a case not as described by the comment?

Reply:
This is to handle a pretty specific edge case: there is only one elder (typically this will be the genesis node) and it's being relocated and there is only one other node that is joined. So this elder demotes itself and promotes the other node. So we go from one elder to also just one elder, but a different one. There is nothing necessarily wrong about this, except it clashes with the way `UpdateBarrier` handles single elder sections. It seemed easier to simply disallow this transition than to try to fix `UpdateBarrier` itself, especially when this whole special handling of single elder section is going to be removed in #2238. 

However, I just realized that with the other change in this PR where we allow relocated current elders as elder candidates, this is no longer necessary because we would never go from A to B, but rather from A to A, B. Going to remove it.